service/src/test/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderServiceTest.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    @Test
+    void shouldFormatC21TemplateDataCorrectlyWhenOnlyC21MandatoryDataIsIncluded() {
+        CaseData caseData = buildCaseData(false);
+
+        Map<String, Object> templateData = createC21OrderService.getC21OrderTemplateData(caseData);
+
+        assertThat(templateData.get("courtName")).isEqualTo("Example Court");
+        assertThat(templateData.get("familyManCaseNumber")).isEqualTo("123");
+        assertThat(templateData.get("orderTitle")).isEqualTo("Order");
+        assertThat(templateData.get("orderDetails")).isEqualTo("Example order details");
+        assertThat(templateData.get("todaysDate")).isEqualTo(
+            dateFormatterService.formatLocalDateToString(time.now().toLocalDate(), FormatStyle.LONG));
+        assertThat(templateData.get("judgeTitleAndName")).isEqualTo("");
+        assertThat(templateData.get("legalAdvisorName")).isEqualTo("");
+        assertThat(templateData.get("children")).isEqualTo(ImmutableList.of());
+    }
+
+    @Test
+    void shouldFormatC21TemplateDataCorrectlyWhenC21TemplateDataIsPopulated() {
+        CaseData caseData = buildCaseData(true);
+
+        Map<String, Object> templateData = createC21OrderService.getC21OrderTemplateData(caseData);
+
+        assertThat(templateData.get("courtName")).isEqualTo("Example Court");
+        assertThat(templateData.get("familyManCaseNumber")).isEqualTo("123");
+        assertThat(templateData.get("orderTitle")).isEqualTo("Example order title");
+        assertThat(templateData.get("orderDetails")).isEqualTo("Example order details");
+        assertThat(templateData.get("todaysDate")).isEqualTo(
+            dateFormatterService.formatLocalDateToString(time.now().toLocalDate(), FormatStyle.LONG));
+        assertThat(templateData.get("judgeTitleAndName")).isEqualTo("His Honour Judge Johnson");
+        assertThat(templateData.get("legalAdvisorName")).isEqualTo("John Clarke");
+        assertThat(templateData.get("children")).isEqualTo(getExpectedChildren());
+    }
+
+    private CaseData buildCaseData(boolean addAllFields) {
+        C21Order.C21OrderBuilder c21OrderBuilder = C21Order.builder()
+            .orderDetails("Example order details");
+
+        CaseData.CaseDataBuilder caseDataBuilder = CaseData.builder()
+            .caseLocalAuthority("example")
+            .familyManCaseNumber("123")
+            .temporaryC21Order(c21OrderBuilder.build());
+
+        if (addAllFields) {
+            caseDataBuilder
+                .temporaryC21Order(c21OrderBuilder
+                    .orderTitle("Example order title")
+                    .build())
+                .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                    .judgeTitle(HIS_HONOUR_JUDGE)
+                    .judgeLastName("Johnson")
+                    .legalAdvisorName("John Clarke")
+                    .build())
+                .children1(createPopulatedChildren())
+                .build();
+        }
+        return caseDataBuilder.build();
+    }
```

Comment:
yeah I am not a fan of a true false private method

Reply:
yeah this is bad, will refactor into two functions